internal/command/extensions/kubernetes/auth.go

```diff
@@ -0,0 +1,115 @@
+package kubernetes
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	fly "github.com/superfly/fly-go"
+	"github.com/superfly/flyctl/gql"
+	"github.com/superfly/flyctl/internal/command"
+	"github.com/superfly/flyctl/internal/command/orgs"
+)
+
+const (
+	execInfoEnv = "KUBERNETES_EXEC_INFO"
+	tokenPrefix = "FlyV1 "
+)
+
+type response struct {
+	APIVersion string `json:"apiVersion"`
+	Kind       string `json:"kind"`
+	Status     status `json:"status"`
+}
+
+type status struct {
+	Token      string `json:"token,omitempty"`
+	Expiration string `json:"expirationTimestamp,omitempty"`
+}
+
+type PartialExecCredential struct {
```

Comment:
I implemented my own version of execCredentials instead of using the one in k8s (found at `clientauthentication.ExecCredentials`) because the k8s version, has `Config` set to `runtime.Object` and I don't know how to turn that into a map.